pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -43,6 +43,52 @@ const (
 	diskTypeUriTemplate = "%s/zones/%s/diskTypes/%s"
 )
 
+type DiskServiceManager interface {
+	// Creates a new persistent disk on GCE with the given disk spec.
+	CreateDiskOnCloudProvider(project string, zone string, disk *compute.Disk) (*compute.Operation, error)
+
+	// Gets the persistent disk from GCE with the given diskName.
+	GetDiskOnCloudProvider(project string, zone string, diskName string) (*compute.Disk, error)
+
+	// Deletes the persistent disk from GCE with the given diskName.
+	DeleteDiskOnCloudProvider(project string, zone string, disk string) (*compute.Operation, error)
+
+	// Waits until GCE reports the given operation in the given zone as done.
+	WaitForZoneOp(op *compute.Operation, zone string, mc *metricContext) error
+}
+
+type GCEServiceManager struct {
+	gce *GCECloud
+}
+
+func (manager *GCEServiceManager) CreateDiskOnCloudProvider(
+    project string,
+    zone string,
+    disk *compute.Disk) (*compute.Operation, error) {
+
+	return manager.gce.service.Disks.Insert(project, zone, disk).Do()
+}
+
+func (manager *GCEServiceManager) GetDiskOnCloudProvider(
+    project string,
+    zone string,
+    diskName string) (*compute.Disk, error) {
+
+	return manager.gce.service.Disks.Get(project, zone, diskName).Do()
+}
+
+func (manager *GCEServiceManager) DeleteDiskOnCloudProvider(
+    project string,
+    zone string,
+    diskName string) (*compute.Operation, error) {
+
+	return manager.gce.service.Disks.Delete(project, zone, diskName).Do()
+}
+
+func (manager *GCEServiceManager) WaitForZoneOp(op *compute.Operation, zone string, mc *metricContext) error {
+	return manager.gce.waitForZoneOp(op, zone, mc)
```

Comment:
It's called inside the disk methods in gce_disks.go. The unit test doesn't issue any actual GCE operations, so this needs to be mocked too.